### PR TITLE
Remove invalid receive/send channel patterns

### PR DESCRIPTION
**Pull request by joelterry**
I hope this isn't disqualifying, but I didn't follow the contributing guidelines: installing Ruby felt a bit extreme for my proposed changes. I did find go.tmLanguage.json in the VSCode source, and made these changes there to test.

I came across these patterns after noticing a highlighting bug when defining a "send channel of channels." 

receive-channel

send-channel

The screenshotted file is unsaved because, upon saving, the second line becomes auto-formatted to match the first. Here I learned for the first time, after checking the [channel section of the spec](https://golang.org/ref/spec#Channel_types), that the receive/send/bidirectional qualification is determined by left associativity rather than prefix/suffix. These two patterns that I recommend removing seem to suggest the opposite (prefix/suffix, must be adjacent). You can see in this [Go Playground example](https://play.golang.org/p/wCfzAGA5ejL) that spacing around the arrows doesn't matter.

Given this rule, "chan <-chan" is super confusing, but that's a separate issue addressed by the formatter.

In summary, my proposal to remove these two channel patterns was prompted by a syntax highlighting bug, but after looking into it, the patterns seem to not be in line with the spec.

### CODE REVIEW

**Comment by jeff-hykin** [comment]
Understandable, I'm fine with removing the "illegal" patterns surrounding channels. I find across many languages "illegal" patterns often do not work as intended. 

Might be a bit before I'm able to push an update to the extension, just FYI

**Comment by jeff-hykin** [comment]
If you could add some test cases to this PR (just some .go files under source/examples) it'll be ready to merge
